w3af/core/ui/api/tests/test_session_scan.py

```diff
@@ -0,0 +1,205 @@
+"""
+test_session_scan.py
+
+Copyright 2015 Andres Riancho
+
+This file is part of w3af, http://w3af.org/ .
+
+w3af is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation version 2 of the License.
+
+w3af is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with w3af; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+"""
+import json
+import requests
+import base64
+
+from w3af.core.ui.api.tests.utils.integration_test import IntegrationTest
+from w3af.core.ui.api.tests.utils.test_profile import (get_expected_vuln_names,
+                                                       get_expected_vuln_urls)
+
+
+class APIScanFromSessionTest(IntegrationTest):
+
+    def test_start_scan_from_session(self):
+        plugin_settings = {
+            "crawl.web_spider": {
+                "only_forward": True,
+                "follow_regex": ".*"},
+            "audit.sqli": {}
+        }
+        target_url = "http://127.0.0.1:8000/audit/sql_injection/"
```

Comment:
Use `get_moth_http` (or similar function, see other tests) to avoid hard-coding the port